modules/apigee_edge_apidocs/tests/src/Functional/ApiDocsAdminTest.php

```diff
@@ -0,0 +1,161 @@
+
+use Drupal\file\Entity\File;
+use Drupal\Tests\BrowserTestBase;
+use Drupal\Tests\TestFileCreationTrait;
+
+/**
+ * Simple test to ensure that main page loads with module enabled.
+ *
+ * @group apigee_edge
+ * @group apigee_edge_apidocs
+ */
+class ApiDocsAdminTest extends BrowserTestBase {
+
+  use TestFileCreationTrait;
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = ['apigee_edge_apidocs', 'block', 'field_ui'];
+
+  /**
+   * A user with permission to administer site configuration.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $adminUser;
+
+  /**
+   * Set up menus and tasks in their regions.
+   *
+   * Since menus and tasks are now blocks, we're required to explicitly set them
+   * to regions.
+   */
+  protected function setupMenus() {
+    $this->drupalPlaceBlock('local_actions_block', ['region' => 'content']);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    // Add the system menu blocks to appropriate regions.
+    $this->setupMenus();
+
+    $this->adminUser = $this->drupalCreateUser([
+      'add apidoc entities',
+      'delete apidoc entities',
+      'edit apidoc entities',
+      'view published apidoc entities',
+      'view unpublished apidoc entities',
+      'administer apidoc entities',
+      'administer apidoc display',
+      'administer apidoc fields',
+      'administer apidoc form display',
+      'access administration pages',
+      'access content',
+    ]);
+    $this->drupalLogin($this->adminUser);
+  }
+
+  /**
+   * Tests that a user can administer API Doc entities.
+   */
+  public function testApiDocAdministration() {
+    $header_selector = 'table .empty';
+
+    $assert = $this->assertSession();
+
+    // Get the API Doc admin page.
+    $this->drupalGet('admin/structure/apidoc');
+
+    // No API Docs yet.
+    $assert->elementTextContains('css', $header_selector, 'There are no API Docs yet.');
+
+    // User can add entity content.
+    $assert->linkExists('Add API Doc');
+    $this->clickLink(t('Add API Doc'));
+
+    // Fields should have proper defaults.
+    $assert->fieldValueEquals('name[0][value]', '');
+    $assert->fieldValueEquals('description[0][value]', '');
+    $assert->fieldValueEquals('status[value]', '1');
+
+    // Create a new spec in site.
+    $file = File::create([
+      'uid' => $this->adminUser->id(),
+      'filename' => 'specA.yml',
+      'uri' => 'public://specA.yml',
+      'filemime' => 'application/octet-stream',
+      'created' => 1,
+      'changed' => 1,
+      'status' => FILE_STATUS_PERMANENT,
+    ]);
+    file_put_contents($file->getFileUri(), "swagger: '2.0'");
+
+    // Save it, inserting a new record.
+    $file->save();
+    $this->assertTrue($file->id() > 0, 'The file was added to the database.');
+
+    $page = $this->getSession()->getPage();
+    $page->fillField('name[0][value]', 'API-Doc-A');
```

Comment:
I would use random values everywhere it is possible instead.

Reply:
Good idea!